template/.github/workflows/ci.yml

```diff
@@ -74,3 +74,43 @@ jobs:
       # NOTE: it runs fine when pushing to Github, but unfortunately not in act.
       # link to issue on act repository: https://github.com/nektos/act/issues/1853
       run: TZ=UTC poetry run invoke build-docs
+
+  run-system-tests:
+    name: Run System Test Suite
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+        cache: 'poetry'
+
+    - name: Install dependencies
+      run: |
+        pip install invoke poetry
+        poetry install
+    - name: Run the system test suite
+      run: poetry run invoke test --suite=system
+
+  run-functional-tests:
+    name: Run Functional Test Suite
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+        cache: 'poetry'
+
+    - name: Install dependencies
+      run: |
+        pip install invoke poetry
+        poetry install
+    - name: Run the functional test suite
+      run: poetry run invoke test --suite=functional --with-deps --verbose
```

Comment:
Issue: running functional tests crashes when launching chromium with the following trace. I raised the issue with playwright team [here](https://github.com/microsoft/playwright/issues/30317).

```
ERROR src/tests/functional/test_humans.py::test_can_get_humans_txt[chromium] - playwright._impl._errors.TargetClosedError: Target page, context or browser has been closed
| Browser logs:
| 
| <launching> /ms-playwright/chromium-1097/chrome-linux/chrome --disable-field-trial-config --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-back-forward-cache --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-component-update --no-default-browser-check --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=ImprovedCookieControls,LazyFrameLoading,GlobalMediaControls,DestroyProfileOnBrowserClose,MediaRouter,DialMediaRouteProvider,AcceptCHFrame,AutoExpandDetailsElement,CertificateTransparencyComponentUpdater,AvoidUnnecessaryBeforeUnloadCheckSync,Translate,HttpsUpgrades,PaintHolding --allow-pre-commit-input --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --force-color-profile=srgb --metrics-recording-only --no-first-run --enable-automation --password-store=basic --use-mock-keychain --no-service-autorun --export-tagged-pdf --disable-search-engine-choice-screen --headless --hide-scrollbars --mute-audio --blink-settings=primaryHoverType=2,availableHoverTypes=2,primaryPointerType=4,availablePointerTypes=4 --no-sandbox --user-data-dir=/tmp/playwright_chromiumdev_profile-Y3rMwA --remote-debugging-pipe --no-startup-window
| <launched> pid=4889
| [pid=4889][err] [0410/085733.867364:ERROR:stack_trace_posix.cc(873)] Failed to parse the contents of /proc/self/maps
| [pid=4889][err] [0410/085734.308809:ERROR:stack_trace_posix.cc(873)] Failed to parse the contents of /proc/self/maps
| [pid=4889][err] [0410/085734.308883:ERROR:stack_trace_posix.cc(873)] Failed to parse the contents of /proc/self/maps
| [pid=4889][err] [0410/085734.340266:FATAL:zygote_main_linux.cc(145)] Check failed: sandbox::ThreadHelpers::IsSingleThreaded(). 
| [pid=4889][err] #0 0x004008474932 base::debug::CollectStackTrace()
| [pid=4889][err] #1 0x004008462573 base::debug::StackTrace::StackTrace()
| [pid=4889][err] #2 0x0040083b9cf3 logging::LogMessage::~LogMessage()
| [pid=4889][err] #3 0x0040083ba7ce logging::LogMessage::~LogMessage()
| [pid=4889][err] #4 0x0040083a3737 logging::CheckError::~CheckError()
| [pid=4889][err] #5 0x00400782ce12 content::ZygoteMain()
| [pid=4889][err] #6 0x004007825db6 content::RunZygote()
| [pid=4889][err] #7 0x004007826c68 content::RunOtherNamedProcessTypeMain()
| [pid=4889][err] #8 0x004007827d88 content::ContentMainRunnerImpl::Run()
| [pid=4889][err] #9 0x0040078256b8 content::RunContentProcess()
| [pid=4889][err] #10 0x0040078258dd content::ContentMain()
| [pid=4889][err] #11 0x004007ebfa1a headless::HeadlessShellMain()
| [pid=4889][err] #12 0x004003ced25a ChromeMain
| [pid=4889][err] #13 0x004011d2dd90 <unknown>
| [pid=4889][err] #14 0x004011d2de40 __libc_start_main
| [pid=4889][err] #15 0x004003ced02a _start
| [pid=4889][err] 
| [pid=4889][err] qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
| [pid=4889][err] [0410/085734.340820:FATAL:zygote_main_linux.cc(145)] Check failed: sandbox::ThreadHelpers::IsSingleThreaded(). 
| [pid=4889][err] #0 0x004008474932 base::debug::CollectStackTrace()
| [pid=4889][err] #1 0x004008462573 base::debug::StackTrace::StackTrace()
| [pid=4889][err] #2 0x0040083b9cf3 logging::LogMessage::~LogMessage()
| [pid=4889][err] #3 0x0040083ba7ce logging::LogMessage::~LogMessage()
| [pid=4889][err] #4 0x0040083a3737 logging::CheckError::~CheckError()
| [pid=4889][err] #5 0x00400782ce12 content::ZygoteMain()
| [pid=4889][err] #6 0x004007825db6 content::RunZygote()
| [pid=4889][err] #7 0x004007826c68 content::RunOtherNamedProcessTypeMain()
| [pid=4889][err] #8 0x004007827d88 content::ContentMainRunnerImpl::Run()
| [pid=4889][err] #9 0x0040078256b8 content::RunContentProcess()
| [pid=4889][err] #10 0x0040078258dd content::ContentMain()
| [pid=4889][err] #11 0x004007ebfa1a headless::HeadlessShellMain()
| [pid=4889][err] #12 0x004003ced25a ChromeMain
| [pid=4889][err] #13 0x004011d2dd90 <unknown>
| [pid=4889][err] #14 0x004011d2de40 __libc_start_main
| [pid=4889][err] #15 0x004003ced02a _start
| [pid=4889][err] 
| [pid=4889][err] qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
| [pid=4889][err] Received signal 11 SEGV_MAPERR 6e696c2d656d6f
| [pid=4889][err] #0 0x004008474932 [0410/085735.002284:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
| [pid=4889][err] base::debug::CollectStackTrace()
| [pid=4889][err] #1 0x004008462573 base::debug::StackTrace::StackTrace()
| [pid=4889][err] #2 0x004008474351 qemu: uncaught target signal 11 (Segmentation fault) - core dumped
```

Reply:
I have just realised we already run the playwright tests in this repository's github actions. See: https://github.com/ackama/django-template/blob/main/.github/workflows/startproject.yml

The template project itself has a single CI actions that runs `startproject` to create new project and then runs the `invoke check` to ensure it has created a viable project.  If you look at that workflow you can see that it installs a number of system dependencies with `apt` to support playwright. This _should_ solve your issues.

I would still keep system and functional tests as separate jobs though as it provides better parallelism.

Reply:
That's a good point. Running the django template ci locally fails with the same error for me. I believe it's either:
- My machine
- some issue in act to setup playwright.

Anyway, I'll update the functional job to include the packages you mentioned.